### PR TITLE
Deprecate ReportGraphConfig

### PR DESCRIPTION
**Pull request by orangejenny**
Followup for https://github.com/dimagi/commcare-hq/pull/16031

Do not merge until migration is run on prod, softlayer, and NIC.

@esoergel / @nickpell

### CODE REVIEW

**Comment by nickpell** [comment]
build error looks real: 

ERROR: test suite for <class 'corehq.apps.app_manager.tests.test_report_config.ReportFiltersSuiteTest'>
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 209, in run
    self.setUp()
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 292, in setUp
    self.setupContext(ancestor)
  File "/vendor/lib/python2.7/site-packages/nose/suite.py", line 315, in setupContext
    try_run(context, names)
  File "/vendor/lib/python2.7/site-packages/nose/util.py", line 471, in try_run
    return func()
  File "/mnt/commcare-hq-ro/corehq/apps/app_manager/tests/test_report_config.py", line 130, in setUpClass
    '7451243209119342931': ReportGraphConfig(
NameError: global name 'ReportGraphConfig' is not defined

**Review by esoergel — APPROVED** [review]
:skull:

File: corehq/apps/app_manager/tests/test_media_suite.py [hunk 1]
```diff
@@ -110,7 +111,14 @@ def test_all_media_report_module(self):
         report._id = 'd3ff18cd83adf4550b35db8d391f6008'
 
         report_app_config = ReportAppConfig(report_id=report._id,
-                                            header={'en': 'CommBugz'})
+                                            header={'en': 'CommBugz'},
+                                            complete_graph_configs = {
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_media_suite.py]: E251 unexpected spaces around keyword / parameter equals

File: corehq/apps/app_manager/tests/test_media_suite.py [hunk 2]
```diff
@@ -110,7 +111,14 @@ def test_all_media_report_module(self):
...
+                                                chart.chart_id: GraphConfiguration(
+                                                    series=[GraphSeries() for c in chart.y_axis_columns],
+                                                )
+                                                for chart in report.charts
+                                            },
+        )
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_media_suite.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/app_manager/tests/test_suite.py [hunk 3]
```diff
@@ -713,7 +715,14 @@ def test_report_module(self):
             header={'en': 'CommBugz'},
             uuid='ip1bjs8xtaejnhfrbzj2r6v1fi6hia4i',
             xpath_description='"report description"',
-            use_xpath_description=True
+            use_xpath_description=True,
+            complete_graph_configs = {
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_suite.py]: E251 unexpected spaces around keyword / parameter equals

File: corehq/apps/app_manager/tests/test_media_suite.py [hunk 4]
```diff
@@ -110,7 +111,14 @@ def test_all_media_report_module(self):
...
         report_app_config = ReportAppConfig(report_id=report._id,
-                                            header={'en': 'CommBugz'})
+                                            header={'en': 'CommBugz'},
+                                            complete_graph_configs={
+                                                chart.chart_id: GraphConfiguration(
+                                                    series=[GraphSeries() for c in chart.y_axis_columns],
+                                                )
+                                                for chart in report.charts
+                                            },
+                                           )
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_media_suite.py]: E124 closing bracket does not match visual indentation

File: corehq/apps/app_manager/tests/test_media_suite.py [hunk 5]
```diff
@@ -110,7 +111,14 @@ def test_all_media_report_module(self):
...
         report_app_config = ReportAppConfig(report_id=report._id,
-                                            header={'en': 'CommBugz'})
+                                            header={'en': 'CommBugz'},
+                                            complete_graph_configs={
+                                                chart.chart_id: GraphConfiguration(
+                                                    series=[GraphSeries() for c in chart.y_axis_columns],
+                                                )
+                                                for chart in report.charts
+                                            },
+                            )
```
Comment (dimagimon) [inline corehq/apps/app_manager/tests/test_media_suite.py]: E124 closing bracket does not match visual indentation

**Comment by orangejenny** [comment]
Migration run on prod and softlayer. Ticket to have it run on NIC: http://manage.dimagi.com/default.asp?254852

**Comment by orangejenny** [comment]
Migration run on NIC. No one one swiss has the flag turned on. Merging.

**Pull request merged**